github-pr.js

```diff
@@ -0,0 +1,113 @@
+"use strict";
+
+/**
+ * Content script that runs in the context of github pull request web pages and
+ * adds "attach to bug" link.
+ */
+
+// Regexp to match against PR title
+var bugRe = /\b(ticket|bug|tracker item|issue)s?:? *([\d ,\+&#and]+)\b/i;
+
+// Base url for attaching a github pr to a bug
+var BASEURL = "https://bugzilla.mozilla.org/attachment.cgi?action=enter&bugid=";
+
+
+/**
+ * Retrieve the PR number from the pull request page.
+ */
+function getPRNum() {
+    // Get the PR number which is like "#4099"
+    var text = document.querySelector('span.gh-header-number').textContent;
+
+    // Peel off the "#" and return
+    return text.substring(1);
+}
+
+
+/**
+ * Retrieve the PR title from the pull request page.
+ */
+function getPRTitle() {
+    return document.querySelector('span.js-issue-title').textContent.trim();
+}
+
+
+/**
+ * Get list of bug ids from PR title
+ */
+function getBugIds(text) {
+    let bugIds = [];
+    let match = bugRe.exec(text);
+    if (match) {
+        match[2].split(/\D+/).forEach(function(bugId) {
+            if (bugId && bugIds.indexOf(bugId) === -1) {
+                bugIds.push(bugId);
```

Comment:
This pattern of picking out unique items is probably better suited with a Set and `.filter()`:

```js
function getBugIds(text) {
  const match = bugRe.exec(text);
  if (match) {
    // if returning a Set object directly isn't ok, you could
    // use Array.from() to convert it into an Array.
    return new Set(match[2].split(/D+/).filter(bugId => !!bugId));
  }
  return new Set();
}
```